source/include/azure_iot_hub_client.h

```diff
@@ -0,0 +1,428 @@
+ * @param[in] pxTransportInterface The #AzureIoTTransportInterface_t to use for the MQTT library.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_Init( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                  const uint8_t * pucHostname,
+                                                  uint32_t ulHostnameLength,
+                                                  const uint8_t * pucDeviceID,
+                                                  uint32_t ulDeviceIDLength,
+                                                  AzureIoTHubClientOptions_t * pxHubClientOptions,
+                                                  uint8_t * pucBuffer,
+                                                  uint32_t ulBufferLength,
+                                                  AzureIoTGetCurrentTimeFunc_t xGetTimeFunction,
+                                                  const AzureIoTTransportInterface_t * pxTransportInterface );
+
+/**
+ * @brief Deinitialize the Azure IoT Hub Client.
+ *
+ * @param xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ */
+void AzureIoTHubClient_Deinit( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle );
+
+/**
+ * @brief Set the symmetric key to use for authentication.
+ * 
+ * @note If using X509 authentication, this is not needed and should not be used.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @param[in] pucSymmetricKey The symmetric key to use for the connection.
+ * @param[in] ulSymmetricKeyLength The length of the \p pucSymmetricKey.
+ * @param[in] xHmacFunction The #AzureIoTGetHMACFunc_t function pointer to a function which computes the HMAC256 over a set of bytes.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_SetSymmetricKey( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                             const uint8_t * pucSymmetricKey,
+                                                             uint32_t ulSymmetricKeyLength,
+                                                             AzureIoTGetHMACFunc_t xHmacFunction );
+
+/**
+ * @brief Connect via MQTT to the IoT Hub endpoint.
+ *
+ * @param[in] xAzureIoTHubClientHandle The #AzureIoTHubClientHandle_t to use for this call.
+ * @param[in] cleanSession A boolean dictating whether to connect with a clean session or not.
+ * @param[in] ulTimeoutMilliseconds The maximum time in milliseconds to wait for a CONNACK.
+ * @return An #AzureIoTHubClientResult_t with the result of the operation.
+ */
+AzureIoTHubClientResult_t AzureIoTHubClient_Connect( AzureIoTHubClientHandle_t xAzureIoTHubClientHandle,
+                                                     bool cleanSession,
```

Comment:
```suggestion
                                                     bool bCleanSession,
```
?

Reply:
I don't think this is a code style.
https://www.freertos.org/FreeRTOS-Coding-Standard-and-Style-Guide.html